scripts/update-component-sha.sh

```diff
@@ -1,10 +1,66 @@
 #!/bin/sh
 set -e
-if [ $# -ne 2 ] ; then
-    echo "Need <OLD> and <NEW> as arguments" >&2
+
+##
+#
+# script to replace hashes in config files
+# see usage() for usage and functionality
+#
+
+function usage() {
+    echo "$0 --<mode> <how-to-find> <new-hash>" >&2
+    echo >&2
+    echo "Avaioable modes: --hash and --image" >&2
+    echo >&2
+    echo "Replace by hash:" >&2
+    echo "	$0 --hash <OLD> <NEW>" >&2
+    echo "	Example: $0 8675309abcdefg abcdef567899" >&2
+    echo "	   Will replace all instances of 8675309abcdefg with abcdef567899" >&2
+    echo >&2
+    echo "Replace by image: $0 --image <IMAGE> <NEW>" >&2
+    echo "	$0 --image <IMAGE> <NEW>" >&2
+    echo "	Example: $0 linuxkit/foo abcdef567899" >&2
+    echo "	   Will tag all instances of linuxkit/foo with abcdef567899" >&2
+    echo >&2
+    echo "By default, for convenience, if no mode is given (--image or --hash), the furst method (--hash) is assumed. Thus the following are equivalent:" >&2
+    echo "	$0 <OLD> <NEW>" >&2
+    echo "	$0 --hash <OLD> <NEW>" >&2
+    echo >&2
+}
+
+
+# backwards compatibility
+if [ $# -eq 2 ]; then
+  set -- "--hash" "$1" "$2"
+fi 
+
+# sufficient arguments
+if [ $# -ne 3 ] ; then
+    usage
     exit 1
 fi
-old=$1
-new=$2
 
-git grep -l "$old" | xargs sed -i -e "s,$old,$new,g"
+
+# which mode?
+case "$1" in
+    --hash)
+        old=$2
+        new=$3
+
+        git grep -l "$old" | xargs sed -i -e "s,$old,$new,g"
+        ;;
+    --image)
+        image=$2
+        hash=$3
+        git grep -l "$image:" | xargs sed -i -e "s,$image:[[:xdigit:]][[:xdigit:]]*,$image:$hash,g"
```

Comment:
So using `-w` works for `--hash` mode, since it always is preceded by a `:` and following by whitespace or end of line (or some other non-word char).

It doesn't work for `--image` mode, since we match with `${image}:`, which is followed by a word-char (the beginning of the hash). I was thinking of dropping the `:`, but then it is likely to match in all sorts of docs and maybe Makefiles, so not a good idea.

May need to just use `-E` with `\b`.